x/btcstaking/keeper/finality_providers.go

```diff
@@ -4,13 +4,48 @@ import (
 	"context"
 	"fmt"
 
+	sdkmath "cosmossdk.io/math"
 	"cosmossdk.io/store/prefix"
 	"github.com/cosmos/cosmos-sdk/runtime"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/babylonlabs-io/babylon/x/btcstaking/types"
 )
 
+// AddFinalityProvider adds the given finality provider to KVStore if it has valid
+// commission and it was not inserted before
+func (k Keeper) AddFinalityProvider(goCtx context.Context, msg *types.MsgCreateFinalityProvider) error {
```

Comment:
Good refactoring! Can `SetFinalityProvider` be made private then?

Reply:
done at https://github.com/babylonlabs-io/babylon/pull/28/commits/1b6e6bafea91db6262dc7b40ae0875ad069023ad